### PR TITLE
build: Optimize macros to reduce compile time

### PR DESCRIPTION
**Pull request by kdy1**
### Testing Instructions

Closes PACK-2571

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
I think it's better to improve compile time gradually.

This reduces compile time from 3m 18s ~ 20s to about 3m

File: crates/turbo-tasks-macros/src/derive/task_input_macro.rs [hunk 1]
```diff
@@ -104,13 +104,15 @@ pub fn derive_task_input(input: TokenStream) -> TokenStream {
...
+            let ident_str = ident.to_string();
+
             (
                 quote! {
                     match value {
                         turbo_tasks::ConcreteTaskInput::List(value) => {
                             let mut #inputs_list_ident = value.iter();
 
-                            let discriminant = #inputs_list_ident.next().ok_or_else(|| anyhow::anyhow!(concat!("missing discriminant for ", stringify!(#ident))))?;
+                            let discriminant = turbo_tasks::macro_helpers::ok_or_else_for_missing_element(#inputs_list_ident.next(), #ident_str)?;
```
Comment (ForsakenHarmony) [inline crates/turbo-tasks-macros/src/derive/task_input_macro.rs:115]: Maybe just turn this into a `let-else`? (same for the other 2)
not the biggest fan of that helper function unless that single thing has a major impact on compile times

**Review by arlyon — APPROVED** [review]
Approving on behalf of repo

**Comment by kdy1** [comment]
Closing as obsolete

**Pull request closed without merging**